nemo_skills/inference/reward_model.py

```diff
@@ -0,0 +1,181 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import importlib
+import json
+import logging
+import sys
+from copy import deepcopy
+from dataclasses import asdict, field
+from pathlib import Path
+
+import hydra
+from tqdm import tqdm
+
+from nemo_skills.inference.server.code_execution_model import ErrorRecoveryConfig, server_params
+from nemo_skills.inference.server.reward_model import get_reward_model
+from nemo_skills.prompt.utils import get_prompt
+from nemo_skills.utils import get_fields_docstring, get_help_message, nested_dataclass, setup_logging
+
+LOG = logging.getLogger(__file__)
+
+
+@nested_dataclass(kw_only=True)
+class RewardModelConfig:
+    """LLM reward model parameters."""
+
+    # Inference server configuration {server_params}
+    server: dict = field(default_factory=dict)
+    # Prompt configuration - path to yaml files
+    prompt_template: str | None = None  # not required for OpenAI server
+    prompt_config: str  # we will fetch it from dataset dir if not provided
+
+    input_file: str | None = None  # Can directly specify an input file, if using a custom dataset
+    output_file: str | None = None  # Where to save the generations if `input_file` is provided
+    input_dir: str | None = (
+        None  # Can specify an input directory, where the file will be inferred output-greedy.jsonl if no seed is provided, and output-rs{{seed}}.jsonl. This pattern is used to match the output files from the `generate` pipeline
```

Comment:
can you move comment to previous line and split it into multiple lines? Otherwise the formatting is not ideal (and multiline comments will still be picked up for cmdline docs)

Reply:
updated in 5a8b5ff3605c4789e652246a64fa8d89f956e8c5